tests/python_package_test/test_dask.py

```diff
@@ -262,6 +262,37 @@ def _unpickle(filepath, serializer):
         raise ValueError(f'Unrecognized serializer type: {serializer}')
 
 
+def _objective_least_squares(y_true, y_pred):
+    grad = y_pred - y_true
+    hess = np.ones(len(y_true))
+    return grad, hess
+
+
+def _objective_logistic_regression(y_true, y_pred):
+    y_pred = 1.0 / (1.0 + np.exp(-y_pred))
+    grad = y_pred - y_true
+    hess = y_pred * (1.0 - y_pred)
+    return grad, hess
+
+
+def _objective_logloss(y_true, y_pred):
+    num_rows = len(y_true)
+    num_class = len(np.unique(y_true))
+    # operate on preds as [num_data, num_classes] matrix
+    y_pred = y_pred.T.reshape(-1, num_class)
+    row_wise_max = np.max(y_pred, axis=1).reshape(num_rows, 1)
+    preds = y_pred - row_wise_max
+    prob = np.exp(preds) / np.sum(np.exp(preds), axis=1).reshape(num_rows, 1)
+    grad_update = np.zeros_like(preds)
+    grad_update[np.arange(num_rows), y_true.astype('int')] = -1.0
+    grad = prob + grad_update
+    hess = 2.0 * prob * (1.0 - prob)
```

Comment:
Per #3256 the hessian gets multiplied by the factor `num_class / (num_class - 1)`
```suggestion
    factor = num_class / (num_class - 1)
    hess = factor * prob * (1 - prob)
```

Reply:
ah ok yeah, thanks for catching that!